crates/nu_plugin_polars/src/lib.rs

```diff
@@ -228,3 +231,19 @@ pub mod test {
         Ok(())
     }
 }
+
+pub(crate) fn handle_panic<F, R>(f: F, span: Span) -> Result<R, ShellError>
+where
+    F: FnOnce() -> Result<R, ShellError>,
+{
+    match catch_unwind(AssertUnwindSafe(f)) {
```

Comment:
I think for the case you use this so far proving unwind-safety could be doable:
- on panic the `LazyFrame` value will discarded and no references
- `Arc` itself is unwind safe
- `LazyFrame` on the other hand `!UnwindSafe`. So the question would be if the unwinding could screw with other dataframes when dropping the lazyframe. Without a narrow reason why we should continue after a panic I am inclined to say that the plugin should still stop and tombstone the variables holding the custom values.


For this generic helper it may be best to [follow the recommendation](https://doc.rust-lang.org/std/panic/struct.AssertUnwindSafe.html#examples) and force the explicit annotation of the captured values as `AssertUnwindSafe` if they are not inherently `UnwindSafe`.

```suggestion
    F: FnOnce() -> Result<R, ShellError> + UnwindSafe,
{
    match catch_unwind(f) {
```

Reply:
No, now that everything is mostly lazy frame based, `polars join` is simply adding to the internal logical plan and returning it as a new lazy frame. The operations don't get applied until collect is called. 

From my testing, unwind did not have any effect on the initial dataframe or other dataframes. This makes sense as collect doesn't effect the initial dataframe, it creates a new one. For the most part, dataframes are treated as immutable objects. Most operations create a new dataframe without modifying the first. 

Polars tends to have a lot of cases where it panics. Itcan be very problematic working with it, have it panic for a random reason, and lose all your work.

Reply:
The suggestion above doesn't work, there are too many issues with trying to get everything underneath to implement. I tried something similar before too with the same issues. I think we have two paths forward:

1. Go with the original pull request suggestion with AssertUnwindSafe
2. Abandon the pull request and let things panic, mark issues like #12732 as won't fix.